pandas/core/arrays/categorical.py

```diff
@@ -1081,20 +1081,26 @@ def remove_unused_categories(self, inplace=False):
             return cat
 
     def map(self, mapper):
-        """Apply mapper function to its categories (not codes).
+        """
+        Map categories (not codes) using input correspondence (a dict,
+        Series, or function).
+
+        Maps the categories to new categories. If the mapping
+        correspondence maps each original category to a different new category
+        the result is a Categorical which has the same order property as
+        the original, otherwise an np.ndarray is returned.
+
+        If a dictionary or Series is used any unmapped category is mapped to
+        NA. Note that if this happens an np.ndarray will be returned.
```

Comment:
Minor but use \`NaN\` instead of NA

Reply:
Please notice that NA is widely used across the pandas documentation:

http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.sum.html#pandas.Series.sum

We asked ourselves whether NA or NaN was more appropriate and decided for NA that seems more used than NaN.
What does the official guideline say on the subject?

Reply:
I don't have the good answer here, that is something we should discuss (in some places I see we use "NA/NaN" ..)

Reply:
So, shall I go for NA or NaN?

Reply:
Use NA when referring to the concept of missing data and `NaN` to refer to the actual missing value. Here you want the latter

Reply:
I modified it.
But please notice that when we have timestamps the missing value is a NaT.

```
In [42]: idx = pd.CategoricalIndex(['a', 'b', 'c'])

In [43]: cat = idx.values

In [44]: type(cat)
Out[44]: pandas.core.arrays.categorical.Categorical

In [45]: cat.map({'a': pd.Timestamp('2000-01-03'), 'b': pd.Timestamp('2000-01-02')})
Out[45]: DatetimeIndex(['2000-01-03', '2000-01-02', 'NaT'], dtype='datetime64[ns]', freq=None)
```

Reply:
>  But please notice that when we have timestamps the missing value is a NaT.

Yes, we need some better guidelines on this, but you can ignore that inconsistency for now